src/Operation.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders;
+
+interface Operation
+{
+    /**
+     * Transform the given set of headers
+     *
+     * @param HeaderBag $headers
+     * @return void
+     */
+    public function modify(HeaderBag $headers);
```

Comment:
I'm not so sure, the two top answers (including the accepted answer) in that SO link you posted are both wrong ;p

Surely this is just a notational change? (since the use case here is identical to a reference)